setup.py

```diff
@@ -188,6 +194,50 @@ def _install_punica() -> bool:
     return bool(int(os.getenv("VLLM_INSTALL_PUNICA_KERNELS", "0")))
 
 
+if _is_cuda():
+
+    # tricky part, nccl 2.19 has a bug that increased memory overhead
+    # of cudagraph. However, pytorch has binary dependencies on nccl 2.19,
+    # simply `pip install nvidia-nccl-cu12==2.18.3` will break pytorch,
+    # so we have to manually download nccl 2.18 and keep the library to
+    #  a secrect place
+
+    # Define the URL of the file and the directory to unzip to
+    file_url = ('https://files.pythonhosted.org/packages/44/6e/'
```

Comment:
Actually I wonder if we can support env var so that we can also decide to load arbitrary .so instead of always downloading our package when we build

Reply:
detecting `env var` during runtime is good.

w.r.t. downloading our package when we build, we have to do this because nccl brought by torch==2.2.0 does not work.

Reply:
how does this solution scale to other backends, like AMD GPU? Does it require another secret .so?